client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx

```diff
@@ -39,7 +41,8 @@ const plans: Step< {
 		return null;
 	}
 
-	const isAllowedToGoBack = isDomainUpsellFlow( flow ) || isNewHostedSiteCreationFlow( flow );
+	const isAllowedToGoBack =
+		isDomainUpsellFlow( flow ) || ( isNewHostedSiteCreationFlow( flow ) && ! isValidWooPartner );
```

Comment:
Thanks, @gavande1, for the changes. They work as expected, it hides the back button for the PTO flow pdt2If-222-p2.

But I think we can take the opportunity to hide the back button for the `/new-hosted-site` flow as well. Basically, removing the `isNewHostedSiteCreationFlow` from this condition. 

@Automattic/quake , do you think it is a good idea to remove the back button when navigating from the https://wordpress.com/hosting/ landing page to the `plans` step? Currently, it navigates back to `/sites` which can be confusing to the user:
![CleanShot 2025-03-31 at 16 30 56@2x](https://github.com/user-attachments/assets/0e26e5a3-7e6c-40dd-8bf4-69ebbed6dd49)

This was originally reported here pdt2If-22d-p2#comment-1441